src/app/qgsmeasuredialog.cpp

```diff
@@ -161,6 +161,31 @@ void QgsMeasureDialog::updateSettings()
   mAreaUnits = QgsProject::instance()->areaUnits();
   mDa.setSourceCrs( mCanvas->mapSettings().destinationCrs(), QgsProject::instance()->transformContext() );
 
+  if ( mMeasureArea )
```

Comment:
Can this duplicate logic be merged with that in QgsMeasureDialog::unitsChanged ? Ie extract this part out to a separate function and call that function both here and in QgsMeasureDialog::unitsChanged.

Reply:
@nyalldawson , done, thanks for the review.